### PR TITLE
#162171040 Implement tracking comment edit history

### PR DESCRIPTION
**Pull request by sudhons**
#### What does this PR do?
- It adds comment edit functionality
- It adds getting previous comment edits functionality

#### Description of Task to be completed?
- Users should be able to edit a comment
- Users should be able to get a comment edit history

#### How should this be manually tested?
- npm run test

#### What are the relevant pivotal tracker stories?
[#162171040](https://www.pivotaltracker.com/story/show/162171040)
#### Any background context you want to add?
Nil
#### Screenshots
Nil

### CODE REVIEW

**Comment by henperi** [comment]
Hello @sudhons, Nice Job so far. I think you miss spelt track as tack in your PR title. Still reviewing your PR. Good job

**Comment by sudhons** [comment]
> Hello @sudhons, Nice Job so far. I think you miss spelt track as tack in your PR title. Still reviewing your PR. Good job

thanks @henperi, It's resolved now

File: test/commentHistory.js [hunk 1]
```diff
@@ -182,9 +241,9 @@ describe('Comment History Tests', () => {
       data.message.should.be.equal('Comment not Found');
     });
     it('should return status code 403 if comment blongs to another user', async () => {
```
Comment (azupatrick0) [inline test/commentHistory.js:243]: @sudhons I think you meant belongs here

File: routes/comment.js [hunk 2]
```diff
@@ -23,8 +21,8 @@ const {
 const { create, list, archive } = CommentController;
 const router = express.Router();
 
-router.get('/:id/comments', checkAuthenticated, checkArticleId, checkArticle, list);
-router.post('/:id/comments', checkAuthenticated, checkCommentContent, checkArticle, create);
+router.get('/:id/comments', checkAuthenticated, checkArticleId, articleExist, list);
```
Comment (azupatrick0) [inline routes/comment.js:24]: @sudhons Nice work so far, but @el-Joft is using /:identifier, try to fix this.

File: routes/comment.js [hunk 3]
```diff
@@ -23,8 +21,8 @@ const {
...
+router.post('/:id/comments', checkAuthenticated, checkCommentContent, articleExist, create);
```
Comment (azupatrick0) [inline routes/comment.js:25]: @sudhons, @el-Joft is using /:identifier, try to fix this.

**Pull request merged**